### DISCUSSION TITLE
Different inverters

### DISCUSSION THREAD

**Discussion by kinichahau2023** [tbnobody/OpenDTU#1317] [Ideas]
Moin allerseits,
da ich dem hervorragenden revers-engineering zum openDTU auch meinen bescheidenen Beitrag geben möchte, starte ich hier als Newbie meine aktuellen Betreibererkenntnisse bzw. Ideen zur Verbesserung.
(Bitte seht mir Fehler nach, da ich hier als Anfänger noch am Anfang stehe - darum auch zunächst mal auf deutsch).

1. Bei Eintragung von 2 verschiedenen Wechselrichtern (WR), von denen jedoch jeweils nur einer aktiv abgefragt/überwacht werden soll (z.B. Hauptgebäude-Anlage und Ferienhaus-Anlage), werden dennoch beide WR vom openDTU abgefragt, obwohl alle 4 Schalter für Datenabruf ausgeschaltet sind:

Log-Data dazu (Ausschnitt, Inverternummern neutralisiert bzw. Fake Nummer):

08:43:25.511 > Fetch inverter: 114112345678
08:43:25.511 > Request SystemConfigPara
08:43:30.524 > Fetch inverter: 114184xxxxxx
08:43:30.524 > Request SystemConfigPara
08:43:30.531 > TX RealTimeRunData Channel: 75 --> 15 84 83 01 70 80 12 08 56 80 0B 00 64 EC 42 12 00 00 00 00 00 00 00 00 A8 CF 9B 
08:43:30.552 > Interrupt received
08:43:30.561 > RX Channel: 40 --> 95 84 83 01 70 84 83 01 70 01 00 01 01 36 00 0E 00 2D 00 0A 00 01 00 00 00 00 8A | -80 dBm
08:43:30.609 > Interrupt received
08:43:30.616 > RX Channel: 23 --> 95 84 83 01 70 84 83 01 70 02 4A 20 00 00 00 00 00 06 00 00 09 19 13 89 00 2B 5A | -80 dBm
08:43:30.713 > Interrupt received
08:43:31.020 > RX Channel: 23 --> 95 84 83 01 70 84 83 01 70 83 00 00 00 02 03 EA 00 BB 00 03 89 05 C9 | -80 dBm
08:43:31.228 > RX Period End
08:43:31.228 > Success
08:43:31.432 > TX SystemConfigPara Channel: 3 --> 15 84 83 01 70 80 12 08 56 80 05 00 64 EC 42 12 00 00 00 00 00 00 00 00 66 C0 54 
08:43:31.636 > Interrupt received
08:43:31.842 > RX Channel: 61 --> 95 84 83 01 70 84 83 01 70 81 00 01 03 E8 00 00 03 E8 FF FF FF FF 01 68 7D F8 F9 | -80 dBm
08:43:32.046 > RX Period End
08:43:32.046 > Success
08:43:35.631 > Fetch inverter: 114112345678
08:43:35.631 > Request SystemConfigPara
08:43:40.542 > Fetch inverter: 114184xxxxxx
08:43:40.559 > TX RealTimeRunData Channel: 23 --> 15 84 83 01 70 80 12 08 56 80 0B 00 64 EC 42 1C 00 00 00 00 00 00 00 00 C8 83 B9 
08:43:40.568 > Interrupt received
08:43:40.581 > RX Chan.....

Da gehen bei "Request SystemConfigPara" 5 sec verloren, die ins Leere laufen, nur weil Inverter Nr. 114112345678 (Fantasie-Nummer) im Menü als 2.Inverter eingetragen ist, aber alle 4 Schalter für Datenabruf ausgeschaltet sind. Erst nach vollständer Löschung des jeweils nicht benötigten WR geht der Abfrageintervall auf seine ca. 5 sec zurück. Ein "Abschalten" bzw. inaktiv setzen des nicht benötigten 2.WR (z.B. Ferienhaus) ist jedoch (auch mittels der 4 Abfrageschalter "Daten abfragen" auf OFF) nicht möglich, da weiterhin auch der inaktive WR abgefragt wird (siehe LOG-Daten)

2. Ähnlich (bzw. dazu sinnvoll ergänzend) verhält es sich mit einer (optionalen?) zweiten WLAN-Eintragung. Dies wurde jedoch bereits früher schon mal angefragt, von jemandem, der die open DTU im Haus vorkonfiguriert, aber dann in der Garage über einen anderen AP laufen lassen wollte. Dasselbe wäre eben auch nützlich, wenn man zwei Anlagen hat (z.B. Hauptanlage plus Ferienhaus) und nur mit einer openDTU beide Anlagen (jeweils nur die aktive Anlage, selektierbar) überwachen möchte durch Mitnahme der openDTU von einer Location zur Anderen. (ich hoffe, es verständlich beschrieben zu haben).

Abschließend noch ein (weniger wichtiger) Vorschlag zur Überschriftenbezeichnung:

openDTU: Titelbezeichnung "Gesamtleistung" ergänzen zu "Gesamtleistung momentan"

Title rename from "Total Power" to "Current Total Power"

**Comment by Tiese** [comment]
Hallo kinichahau2023,

mit der Abfrage hat du recht - ist bei mir genauso.

Und mit der Bezeichnung meinst du das bestimmt so...
![Bildschirmfoto 2023-09-08 um 12 39 30](https://github.com/tbnobody/OpenDTU/assets/141029037/578392f9-9a3f-4ae3-8bea-1c9a55dbbe22)

**Reply by kinichahau2023** [reply]
Hallo Tiese,
Ja die (bei mir dritte/letzte) grüne Überschrift... 

![20230908_152121](https://github.com/tbnobody/OpenDTU/assets/144427583/12bc8c7c-6147-4a3c-80bf-eb759cee85a3)

Wobei ich kein "Kostenersparnis"-Segment in meinem Display habe... woher hast Du das?

**Reply by kinichahau2023** [reply]
sorry  - habs gerade gefunden mit der Kostenersparnis ... :)

**Reply by Tiese** [reply]
Hier ist die Version mit der "Gesamtleistung momentan"...
[firmware.bin.zip](https://github.com/tbnobody/OpenDTU/files/12561032/firmware.bin.zip)

Und wie du siehst, sind auch die Meldungen in deutsch (lässt sich aber "noch" nicht wechseln).

**Reply by kinichahau2023** [reply]
Hi,
Besten Dank.
Frage 1 dazu: genauso OTA zu handeln wie das generic.bin?
Anmerkung: bin etwas unentschlossen, denn solange Deine Verbesserungen/Änderungen nicht auch im generic.bin implementiert sind, ist es beim nächsten generic.bin-update alles wieder weg ... 
Muss ich noch mal überlegen... 
Trotzdem super, ich bin leider noch nicht so weit, mir solche Änderungen selbst zusammenzustellen. Aber kommt Zeit, kommt auch das.

**Reply by Tiese** [reply]
Du kannst ja jeder Zeit die offizielle Version wieder aufspielen!
